statsd/receiver.go

```diff
@@ -188,10 +189,15 @@ func (mr *metricReceiver) getAdditionalTags(addr string) types.Tags {
 
 // ParseLine with lexer impl.
 func (mr *metricReceiver) parseLine(line []byte) (*types.Metric, error) {
+	if line == nil {
+		return nil, errors.New("Error parsing line. Byte array is nil")
+	}
+
 	llen := len(line)
 	if llen == 0 {
-		return nil, nil
+		return nil, errors.New("Error parsing line. Byte array is empty")
```

Comment:
I don't think it is possible to hit this branch in the current codebase - there is a check before this function is invoked:
```go
if len(line) > 1 {
    metric, err := mr.parseLine(line)
```
I agree the code is not ideal (a redundant check inside of the function) and I've cleanup it up already in another branch.